src/lib/expansion/expansion-panel-header.html

```diff
@@ -3,5 +3,13 @@
   <ng-content select="mat-panel-description"></ng-content>
   <ng-content></ng-content>
 </span>
-<span [@indicatorRotate]="_getExpandedState()" *ngIf="_showToggle()"
-      class="mat-expansion-indicator"></span>
+<span [@indicatorRotate]="_getExpandedState()"
```

Comment:
In general we try to avoid proxying many attributes like this. An alternate approach can be to have an `ng-content` in here that allows consumers to project their own toggle. If something is projected into it, we can then remove the default toggle.

Reply:
I think the problem of using ng-content in this case is that the element which performs toggling functionality will probably not be a "toggle" button (because otherwise it doesn't make too much sense to hide the toggle button in the first place), but something that is meaningful inside the header. For example, we could have some text in the header which can be toggled by keyboard to expand or collapse the panel. The text will be part of the elements projected in the last `<ng-content>` of `mat-content` span and therefore it may not be appended at the end of the header.

Reply:
The `ng-content` can have a selector so that the button is picked out from all the other content around it.

Reply:
Correct me if I'm wrong. I don't think we know exactly where the `ng-content` selector should be placed in the header. The button could be located in the middle of the header DOM, e.g. 

```
<mat-expansion-panel>
  <mat-expansion-panel-header>
    <span>Some text</span>
    <span>Some text that should toggle the panel</span>
    <span>Some other text</span>
  </mat-expansion-panel-header>
  <ng-template matExpansionPanelContent>
    ...
  </ng-template>
</mat-expansion-panel>
```

It doesn't seem to be possible to project the second `span` into the header without modifying the UI layout.

Reply:
@crisbeto Any further thoughts on how I should proceed? An alternative that I could think of is not showing the toggle button at all if the header is not a button, because the parent component will provide an interactive element for expanding/collapsing the card.

Reply:
Sorry for the delayed response, I was on vacation for a while. 

I chatted with the rest of the team about this and we came up with another approach we like. We can introduce a directive like `<button matExpansionPanelToggle>`. The header would use `@ContentChild` to check for the presence of `matExpansionPanelToggle`. When it's there, the `role` and other `aria-` attributes is removed from the header. The nice thing here is that the `matExpansionPanelToggle` can then have the appropriate ARIA treatment for that button so that users don't have to do it themselves.

How does that sound to you?

Reply:
Oh, I commented on the wrong chain, but the content still applies.